onnxruntime/core/providers/cuda/tensor/quantize_linear.cc

```diff
@@ -183,33 +422,60 @@ REGISTER_Q_KERNEL_TYPED_10_12(uint8_t)
 REGISTER_Q_KERNEL_TYPED_13_18(int8_t)
 REGISTER_Q_KERNEL_TYPED_13_18(uint8_t)
 
-#define REGISTER_Q_KERNEL_TYPED_19(T)                                     \
-  ONNX_OPERATOR_TWO_TYPED_KERNEL_EX(                                      \
+#define REGISTER_Q_KERNEL_TYPED_19_20(T)                                  \
+  ONNX_OPERATOR_VERSIONED_TWO_TYPED_KERNEL_EX(                            \
       QuantizeLinear,                                                     \
       kOnnxDomain,                                                        \
-      19,                                                                 \
+      19, 20,                                                             \
       T, float,                                                           \
       kCudaExecutionProvider,                                             \
       (*KernelDefBuilder::Create())                                       \
           .TypeConstraint("T1", DataTypeImpl::GetTensorType<float>())     \
           .TypeConstraint("T2", DataTypeImpl::GetTensorType<T>()),        \
       QuantizeLinear<T, float>);                                          \
-  ONNX_OPERATOR_TWO_TYPED_KERNEL_EX(                                      \
+  ONNX_OPERATOR_VERSIONED_TWO_TYPED_KERNEL_EX(                            \
       QuantizeLinear,                                                     \
       kOnnxDomain,                                                        \
-      19,                                                                 \
+      19, 20,                                                             \
       T, MLFloat16,                                                       \
       kCudaExecutionProvider,                                             \
       (*KernelDefBuilder::Create())                                       \
           .TypeConstraint("T1", DataTypeImpl::GetTensorType<MLFloat16>()) \
           .TypeConstraint("T2", DataTypeImpl::GetTensorType<T>()),        \
       QuantizeLinear<T, MLFloat16>);
 
-REGISTER_Q_KERNEL_TYPED_19(int8_t)
-REGISTER_Q_KERNEL_TYPED_19(uint8_t)
+REGISTER_Q_KERNEL_TYPED_19_20(int8_t)
+REGISTER_Q_KERNEL_TYPED_19_20(uint8_t)
+#if !defined(DISABLE_FLOAT8_TYPES)
+REGISTER_Q_KERNEL_TYPED_19_20(Float8E4M3FN)
+REGISTER_Q_KERNEL_TYPED_19_20(Float8E5M2)
+#endif
+
+#define REGISTER_Q_KERNEL_TYPED_21(T, U)                           \
```

Comment:
>REGISTER_Q_KERNEL_TYPED_21 [](http://example.com/codeflow?start=8&length=26)

nit: keep the register consistent. Either use the old way (unroll float and mlfloat16), or the new way.

Reply:
updated to new way